zad2/warriors.rb

```diff
@@ -1,12 +1,103 @@
-class Warrior
+# Mateusz Zieba
+# My solution with extended fight mechanism
+
+class Character
+  attr_accessor :health_points, :multiplier
+  attr_reader :name, :level
+  
   def initialize(name:, level:)
+    @name = name
+    @level = level  
+    @multiplier = 20
+    @health_points = @level * @multiplier  
   end
-
+  
+  def card
+    "#{@name} | HP: #{@health_points} | level: #{@level}" 
+  end
+  
+  def roll
+    [*1..12].sample
+  end
+  
   def strength
+    roll + @level
+  end
+
+  def regen_health
+    @health_points = @multiplier * level
+  end
+
+  def attack(victim)
```

Comment:
`attack` zwraca `true` / `false` co jest trochę mało intuicyjne i trzeba zerknąć w kod żeby zobaczyć co oznaczają te booleany

Reply:
Rzeczywiście, zastanowię się nad tym. Zależało mi na tym, aby pętla odpowiadająca za walkę dwóch postaci była prosta i czytelna, ale na pewno można to rozwiązać tak, by było bardziej intuicyjne :)